tests/js/session-tests.js

```diff
@@ -237,6 +237,114 @@ module.exports = {
             });
     },
 
+    testProgressNotificationsForRealmOpen() {
+        if (!isNodeProccess) {
+            return Promise.resolve();
+        }
+
+        const username = uuid();
+        const realmName = uuid();
+        const expectedObjectsCount = 3;
+
+        let tmpDir = tmp.dirSync();
```

Comment:
Seems like it's about time we abstract all of this into a `runInSeparateProcess('download-realm-api.js')` helper call or something like that?